src/Providers/LIAuthServiceProvider.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace LiveIntent\LaravelCommon\Providers;
+
+use Illuminate\Auth\RequestGuard;
+use Illuminate\Support\Facades\Auth;
+use LiveIntent\LaravelCommon\Auth\Guards\LITokenGuard;
+use Illuminate\Foundation\Support\Providers\AuthServiceProvider as ServiceProvider;
+
+class LIAuthServiceProvider extends ServiceProvider
+{
+    public function __construct($app)
+    {
+        parent::__construct($app);
+        $this->registerEnvironmentVariables();
+    }
+
+    /**
+     * Register any authentication / authorization services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        $this->registerCustomGuards();
+        $this->registerPolicies();
+    }
+
+    protected function registerEnvironmentVariables()
+    {
+        if (config('auth.li_token.keys.public') == null) {
+            config(['auth.li_token.keys.public' => env('TESSELLATE_PUBLIC_KEY') ]);
+        }
+
+        if (config('auth.guards.li_token') == null) {
+            config(['auth.guards.li_token' => [
+                'driver' => 'li_token',
+                'provider' => 'users',
+            ]]);
+        }
+
+        $this->policies = config('auth.li_token.policies', []);
+    }
+
+    protected function registerCustomGuards()
+    {
+        Auth::extend('li_token', function ($app, $_name, array $config) {
+            throw_if($config['provider'] == null);
+
+            return new RequestGuard(function ($request) use ($config) {
+                return (new LITokenGuard(
+                    Auth::createUserProvider($config['provider']),
+                ))->user($request);
+            }, $app['request']);
+        });
+
+        // Causes issues with an infinite loop, cannot set `li_token` as default guard with current implementation
+//        Auth::shouldUse('li_token');
```

Comment:
When I attempted to use this to set the default auth guard, (or even manually configuring the file and putting `'li_token'` as the default) this resulted in an infinite loop of authenticating the token. Perhaps there's something wrong with the way RequestGuards work. 

I was going to change `LITokenGuard` into an actual Guard object, but figured I'd push this up as is for now. Let me know if you want me to try that.

Reply:
This package should only expose the guard as being available to use, but it should not set the default guard in the consuming app IMO. Remember, this package is meant to provide a set of tools that can be used when desired, but we want to make sure we give the flexibility and choice to the consumer and don't introduce too many hidden "behavior" changes to consuming apps.

To set the default guard I would expect a consuming app to set the value here https://github.com/LiveIntent/tpl-laravel/blob/main/config/auth.php#L17 to li_token.

Reply:
Sure, that makes sense. This line checks if the default guard is set, and if it isn't, only then will it set it to `li_token`. However, setting the default guard through that mechanism or as seen in the `auth.php` on line 17 (as per your link) ends up causing an infinite loop even though it recognizes that there is only one guard.